pkg/validation/limits.go

```diff
@@ -478,6 +480,18 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 		false,
 		"Enable persistence of patterns detected at ingest. When enabled, patterns for pushed streams will be written back into Loki as a special __pattern__ stream.",
 	)
+	f.DurationVar(
+		(*time.Duration)(&l.PatternPersistenceGranularity),
+		"limits.pattern-persistence-granularity",
+		0,
+		"The time granularity for persisting patterns. Controls how many data points are written when patterns are flushed. Set to 0 to use the default from the pattern ingester configuration.",
+	)
```

Comment:
I would follow the same pattern we use elsewhere in this file

```suggestion
    _ = l.PatternPersistenceGranularity.Set("0s") # not even needed
	f.Var(&l.PatternPersistenceGranularity, "limits.pattern-persistence-granularity", "The time granularity for persisting patterns. Controls how many data points are written when patterns are flushed. Set to 0 to use the default from the pattern ingester configuration.")
```